lib/tokenlist.cpp

```diff
@@ -779,7 +779,7 @@ static void compileTerm(Token *&tok, AST_state& state)
             prev = prev->link()->previous();
         if (Token::simpleMatch(tok->link(),"} [")) {
             tok = tok->next();
-        } else if (state.cpp && iscpp11init(tok)) {
+        } else if ((state.cpp && iscpp11init(tok)) || Token::simpleMatch(tok->previous(), "] {")) {
```

Comment:
Shouldn't the new check be inside `iscpp11init()`?

Reply:
@chrchr-github Your example is not valid. The code we want to handle is for example:

	auto p = new int*[2]{new int, new int};

That specific code is not valid in C but imho something similar is possible in C. This is for instance valid C code:

```
struct S {int x; int y;};

struct S s1;
struct S s2;

int main() {
	(struct S*[2]){&s1, &s2};
	return 0;
}
```
I don't know if something that is more similar can be written in valid C code.